Modules/socketmodule.c

```diff
@@ -625,16 +625,16 @@ set_gaierror(int error)
 
 #ifdef __VMS
 /* Function to send in segments */
-static int
-sendsegmented(int sock_fd, char *buf, int len, int flags)
+static Py_ssize_t
+sendsegmented(int sock_fd, char *buf, Py_ssize_t len, int flags)
 {
     int n = 0;
-    int remaining = len;
+    Py_ssize_t remaining = len;
 
     while (remaining > 0) {
         unsigned int segment;
 
-        segment = (remaining >= SEGMENT_SIZE ? SEGMENT_SIZE : remaining);
+        segment = ((size_t)remaining >= SEGMENT_SIZE ? SEGMENT_SIZE : (unsigned int) remaining);
```

Comment:
Wait, is this cast safe? In C99 ssize_t can be an unsigned long, unless I am missing something

Reply:
SEGMENT_SIZE is 32 KiB: remaining is only casted for values between [0; 32 KiB]:

```
#define SEGMENT_SIZE (32 * 1024 -1)
```